FC-1157 Provisioning (#3)
Templum/developers-community: pages/Documents/DeveloperTools/LivePersonFunctions/provisioning.md

indicator: both
---

To leverage the full functionality of the LivePerson Functions platform we require to perform some provisioning tasks. This process needs to be performed by a Conversational Cloud Admin and it happens during the initial agreement to the terms of use. Furthermore, we may perform partial provisioning in the future if new features require them. This process will happen in the background during login to the UI.

> <img src="img/faas-terms-of-use.png" alt="Functions: Terms of Use" style="width:100%;"/>

Reviewer: @eplazaso pushes are changed. You can now rebase it on master and make the change (all images are located inside functions folder)

> <img src="img/faas-terms-of-use.png" alt="Functions: Terms of Use" style="width:100%;"/>

Author: Added proper image in the right folder

Following provisioning tasks will be performed:

* Activating the LivePerson Functions feature on your account.
* [Installing default secrets](liveperson-functions-development-storing-secrets.html) to interact with other LivePerson APIs
* [Whitelisting default domains](liveperson-functions-development-whitelisting-domains.html) of internal LivePerson APIs

### Provisioning secrets

> Functions interaction with other LivePerson APIs need to get authenticated therefore key/secrets are required. During provisioning, the following secrets are added to your account by default.

Reviewer 2: ```suggestion
Interaction between Functions and other LivePerson APIs needs to get authenticated. Therefore, special key/secret pairs are added to your account during provisioning. These secrets are as follows:
```

> * **lp-faas-default-app-key**: This secret is required and used by the LP Client to access LivePerson Apis, which is available in the Toolbelt.

Reviewer: Missing link to toolbelt

* **lp-faas-orchestrator-app-key**: This secret is required and used for the Orchestrator Feature, which is available in the Toolbelt.

> <div class="notice">In order to facilitate functions development, Toolbelt clients for internal APIs make use of those secrets, in this way the functions developers don't need to implement any authentication method for those APIs.</div>

Reviewer 2: ```suggestion
<div class="notice">In order to ease functions development, Toolbelt clients for internal APIs make use of those secrets, in this way the functions developers don't need to implement any authentication method for those APIs.</div>
```

    <td style="text-align: center; vertical-align: middle;" >Read</td>
  </tr>
  <tr>
    <td style="text-align: center; vertical-align: middle;" >Predefined Categories API</td>
    <td style="text-align: center; vertical-align: middle;" >Read</td>
  </tr>
  <tr>
    <td style="text-align: center; vertical-align: middle;" >Unified Automatic Messages API </td>
    <td style="text-align: center; vertical-align: middle;" >Read</td>
  </tr>
  <tr>
    <td style="text-align: center; vertical-align: middle;" >Agent Status Reason API</td>
    <td style="text-align: center; vertical-align: middle;" >Read</td>
  </tr>
  </tbody>
</table>

#### lp-faas-orchestrator-app-key privileges

<table class="thinner" style="width: 100%">
  <thead>
    <tr>
      <th  style="text-align: center; vertical-align: middle;">API</th>
      <th  style="text-align: center; vertical-align: middle;">Privileges</th>
    </tr>
  </thead>
  <tbody>
  <tr>
    <td style="text-align: center; vertical-align: middle;" >Functions API</td>
    <td style="text-align: center; vertical-align: middle;" >Invoke</td>
  </tr> 
  </tbody>
</table>

### Default whitelisted domains

> The [Toolbelt](function-as-a-service-foundations-features.html) clients use the below domains to call internal LivePerson APIs. Those APIs domains are whitelisted during provisioning and are accessible from the functions.

Reviewer 2: ```suggestion
The [Toolbelt](function-as-a-service-foundations-features.html) clients use the domains below to call internal LivePerson APIs. These API domains are whitelisted during provisioning and are accessible from your functions.
```

<table class="thinner" style="width: 100%">
<thead>
  <tr>
    <th  style="text-align: center; vertical-align: middle;">Domain Name</th>
    <th  style="text-align: center; vertical-align: middle;">API Name</th>
    <th  style="text-align: center; vertical-align: middle;">Description</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td>faaSGW</td>
    <td style="text-align: center; vertical-align: middle;" >Liveperson Functions Gateway</td>
    <td style="text-align: center; vertical-align: middle;" >Required for Orchestrator FN client in order to invoke other functions.</td>
  </tr> 
  <tr>
    <td>msgHist</td>
    <td style="text-align: center; vertical-align: middle;" >Consumer Messaging History API</td>

>     <td style="text-align: center; vertical-align: middle;" >Retrieve one consumer's messaging data for the last 13 months.</td>

Reviewer 2: ```suggestion
    <td style="text-align: center; vertical-align: middle;" >Retrieve a single consumer's messaging data for the last 13 months.</td>
```

    <td style="text-align: center; vertical-align: middle;" >(Only to retrieve data) Includes a set of different APIs for you LE account management. E.g Predefined Content, Automatic Messages, Workdays... </td>
  </tr> 
  <tr>
    <td>accountConfigReadWrite</td>
    <td style="text-align: center; vertical-align: middle;" >Agent Groups API, Skills API, Users API, LOBs API, Profiles API, Automatic Messages API, Unified Automatic Messages API, Predefined Content API, Predefined Categories API, Agent Status Reason API, Special Occasions API, Workdays API, Audit Trail API</td>
    <td style="text-align: center; vertical-align: middle;" >(Create/retrieve data) Includes a set of different APIs for you LE account management. E.g Predefined Content, Automatic Messages, Workdays...</td>
  </tr>
  <tr>
    <td>asyncMessaging</td>
    <td style="text-align: center; vertical-align: middle;" >Messaging Window API, Connector API and Shift Status API</td>

>     <td style="text-align: center; vertical-align: middle;" >Build custom visitor experiences for messaging. Build a visitor chat experience outside of the web browser environment. Create a session, read agent lines, send your own lines, etc</td>

Reviewer 2: ```suggestion
    <td style="text-align: center; vertical-align: middle;" >Build a custom visitor experiences for messaging or a visitor chat experience outside of the web browser environment. Create a session, read agent lines, send your own lines, etc.</td>
```

  </tr>
  <tr>
    <td>asyncMessagingEnt</td>
    <td style="text-align: center; vertical-align: middle;" >Messaging Window API, Connector API and Shift Status API</td>

>     <td style="text-align: center; vertical-align: middle;" >Build custom visitor experiences for messaging. Build a visitor chat experience outside of the web browser environment. Create a session, read agent lines, send your own lines, etc</td>

Reviewer 2: same as asyncMessaging (?) not sure if necessary or whats the difference here.

Author: I was haven't really found the difference, I just found that usually both domains are used for the same APIs but it can change for some customers, but don't know more... https://confluence.liveperson.com/display/SOCIAL/Development+on+UMS

Reviewer 2: 👍

  </tr>
  <tr>
    <td>engHistDomain</td>
    <td style="text-align: center; vertical-align: middle;" >Engagement History API</td>
    <td style="text-align: center; vertical-align: middle;" >Historical download of chat conversations</td>
  </tr>
  <tr>
    <td>rtbf</td>
    <td style="text-align: center; vertical-align: middle;" >Personal Data Deletion API</td>

>     <td style="text-align: center; vertical-align: middle;" >Keep track of and make requests for data deletion to comply with GDPR</td>

Reviewer 2: ```suggestion
    <td style="text-align: center; vertical-align: middle;" >Keep track of and make requests for data deletion to comply with GDPR.</td>
```

  </tr>
  <tr>
    <td>smt</td>
    <td style="text-align: center; vertical-align: middle;" >IVR Engagement API, Validate Engagement API and Visit Information API</td>
    <td style="text-align: center; vertical-align: middle;" >APIs for engagement data and visitors information</td>
  </tr>
  <tr>
    <td>pusher</td>
    <td style="text-align: center; vertical-align: middle;" >Push Notification Service</td>

>     <td style="text-align: center; vertical-align: middle;" >Work and register to LivePerson's push notification service</td>

Reviewer 2: ```suggestion
    <td style="text-align: center; vertical-align: middle;" >Work and register to LivePerson's push notification service.</td>
```

indicator: both
---

> In order to easily integrate LivePerson Functions, we will allow Conversational Cloud Admins to provision your account when entering LivePerson Functions. Provisioning will be triggered after you agree to our terms of use, if provisioning fails double-check the privileges on your user:

Reviewer: ```suggestion
To be able to leverage the full functionality of the LivePerson Functions platform we require to perform some provisioning tasks. This process needs to be performed by a Conversational Cloud Admin and it happens during the initial agreement to the terms of use. Further additional we may perform partial provisioning in the future if new features require them. This process will happen in the background during login to the UI.

<img src="img/faas-terms-of-use.png" alt="FaaS Terms of Use" style="width:70%;"/>

Following provisioning tasks will be performed:
```

Reviewer: Maybe width can be also set to 100%, check how it looks with 70%

> <img src="img/faas-terms-of-use.png" alt="FaaS Terms of Use" style="width:100%;"/>

Reviewer: ```suggestion
```

Author: so we should remove the image? i thin it looks better with that screenshot

> Once agreed, we will automatically set up your account to easily integrate with the rest of the LivePerson ecosystem including:

Reviewer: ```suggestion
```

* Activating the LivePerson Functions feature on your account.
* [Installing default secrets](liveperson-functions-development-storing-secrets.html) to interact with other LivePerson APIs
* [Whitelisting default domains](liveperson-functions-development-whitelisting-domains.html) of internal LivePerson APIs

### Provisioning secrets

Functions interaction with other LivePerson APIs need to get authenticated therefore key/secrets are required. During provisioning, the following secrets are added to your account by default.

> * **lp-faas-default-app-key**: Required for internal LivePerson API calls.

Reviewer: ```suggestion
* **lp-faas-default-app-key**: This secret is required and used by the LP Client to access LivePerson Apis, which is available in the Toolbelt.
```

Reviewer: The link should go to LP CLient.

> * **Lp-faas-orchestrator-app-key**: Required for [Toolbelt](function-as-a-service-foundations-features.html) OrchestratorFN client.

Reviewer: ```suggestion
* **lp-faas-orchestrator-app-key**: Required for [Toolbelt](function-as-a-service-foundations-features.html) OrchestratorFN client.
```

> * **Lp-faas-orchestrator-app-key**: Required for [Toolbelt](function-as-a-service-foundations-features.html) OrchestratorFN client.

Reviewer: ```suggestion
* **Lp-faas-orchestrator-app-key**: This secret is required and used for the Orchestrator Feature, which is available in the Toolbelt.
```

Reviewer: Link should be on the Orchestrator Feature

Author: But is also a defautlt key, i think it should be in both

Reviewer: But we have no link at all here

Reviewer: Please add a link from `Orchestrator Feature` to the Toolbelt (later we will than need to adjust it to the correct subpage)

  </tr>
  <tr>
    <td style="text-align: center; vertical-align: middle;" >API</td>
    <td style="text-align: center; vertical-align: middle;" >external-engagement-api-signed</td>
  </tr>
  <tr>
    <td style="text-align: center; vertical-align: middle;" >API</td>
    <td style="text-align: center; vertical-align: middle;" >ac.canned.response.ext.r</td>
  </tr>
  <tr>
    <td style="text-align: center; vertical-align: middle;" >API</td>
    <td style="text-align: center; vertical-align: middle;" >ac.auto.message.ext.r</td>
  </tr>
  <tr>
    <td style="text-align: center; vertical-align: middle;" >API</td>
    <td style="text-align: center; vertical-align: middle;" >ac.categories.read.external</td>
  </tr>
  <tr>
    <td style="text-align: center; vertical-align: middle;" >API</td>
    <td style="text-align: center; vertical-align: middle;" >ac.messaging.auto.message.ext.r</td>
  </tr>
  <tr>
    <td style="text-align: center; vertical-align: middle;" >API</td>
    <td style="text-align: center; vertical-align: middle;" >ac.multiple.agent.status.read.external</td>
  </tr>
  <tr>
    <td style="text-align: center; vertical-align: middle;" >VISIBLE_TO_ACCOUNT_ADMIN</td>

>     <td style="text-align: center; vertical-align: middle;" >true</td>

Reviewer: This is nothing for the users of the platform, it is internal knowledge.... Remove it and replace it with easier to understand API's names

  </tr>
  </tbody>
</table>

> #### Lp-faas-orchestrator-app-key privileges

Reviewer: ```suggestion
#### lp-faas-orchestrator-app-key privileges
```

<table class="thinner" style="width: 100%">
  <thead>
    <tr>
      <th  style="text-align: center; vertical-align: middle;">TYPE</th>
      <th  style="text-align: center; vertical-align: middle;">Privileges</th>
    </tr>
  </thead>
  <tbody>
  <tr>
    <td style="text-align: center; vertical-align: middle;" >API</td>
    <td style="text-align: center; vertical-align: middle;" >faas.lambda.invoke.external</td>
  </tr> 

>   </tbody>

Reviewer: Same as the above

</table>

### Default whitelisted domains

> The [Toolbelt](function-as-a-service-foundations-features.html) clients use the bellow domains to call internal LivePerson APIs. Those APIs domains are whitelisted during provisioning and are accessible from the functions.

Reviewer: ```suggestion
The [Toolbelt](function-as-a-service-foundations-features.html) clients use the below domains to call internal LivePerson APIs. Those APIs domains are whitelisted during provisioning and are accessible from the functions.
```
